### PR TITLE
refactor: math and adding eth tests

### PR DESCRIPTION
**Pull request by kupermind**
-  Math and adding eth tests.

### CODE REVIEW

File: contracts/BuyBackBurner.sol [hunk 1]
```diff
@@ -115,7 +116,8 @@ abstract contract BuyBackBurner {
         uint160 sqrtPriceX96 = TickMath.getSqrtRatioAtTick(averageTick);
 
         // Calculate the price using the sqrtPriceX96
-        price = (uint256(sqrtPriceX96) * uint256(sqrtPriceX96)) / (1 << 192);
+        // Max result is uint160 * uint160 == uint320, not to overflow: 320 - 256 = 64 (2^64)
```
Comment (kupermind) [inline contracts/BuyBackBurner.sol:119]: Fixed math from overflows and minimal values converting to zero.

File: contracts/BuyBackBurner.sol [hunk 2]
```diff
@@ -225,13 +227,14 @@ abstract contract BuyBackBurner {
         // Check TWAP or historical data
         uint256 twapPrice = _getTwapFromOracle(pool);
         // Get instant price
-        uint256 instantPrice = (uint256(sqrtPriceX96) * uint256(sqrtPriceX96)) / (1 << 192);
+        // Max result is uint160 * uint160 == uint320, not to overflow: 320 - 256 = 64 (2^64)
```
Comment (kupermind) [inline contracts/BuyBackBurner.sol:230]: This is in line with UniV3

File: contracts/BuyBackBurnerBalancer.sol [hunk 3]
```diff
@@ -37,8 +37,8 @@ interface IERC20 {
     function approve(address spender, uint256 amount) external returns (bool);
 }
 
-/// @title BuyBackBurnerBase - BuyBackBurner implementation contract for Base
-contract BuyBackBurnerBase is BuyBackBurner {
+/// @title BuyBackBurnerBalancer - BuyBackBurner implementation contract for interaction with Balancer
+contract BuyBackBurnerBalancer is BuyBackBurner {
```
Comment (kupermind) [inline contracts/BuyBackBurnerBalancer.sol:41]: This name makes much more sense

File: contracts/MemeFactory.sol [hunk 4]
```diff
@@ -315,7 +315,7 @@ abstract contract MemeFactory {
         bytes memory payload = abi.encodePacked(type(Meme).creationCode, abi.encode(name, symbol, DECIMALS, totalSupply));
         // solhint-disable-next-line no-inline-assembly
         assembly {
-            memeToken := create2(0x0, add(0x20, payload), mload(payload), memeNonce) // TOFIX: revert to randomNonce
+            memeToken := create2(0x0, add(0x20, payload), mload(payload), randomNonce)
```
Comment (kupermind) [inline contracts/MemeFactory.sol:318]: This is key, needs to stay this way and revert to testing from now on.

**Review by DavidMinarsch — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**